tests/test_favorite.py

```diff
@@ -0,0 +1,239 @@
+import json
+from http.client import OK
+from http.client import NOT_FOUND
+from http.client import CREATED
+from http.client import BAD_REQUEST
+from http.client import NO_CONTENT
+from peewee import SqliteDatabase
+from models import Item, User, Favorites
+from app import app
+import uuid
+
+
+def create_user(id_user, number):
+    return User.create(
+        uuid=id_user,
+        first_name='{}{}'.format('first_name_', number),
+        last_name='{}{}'.format('last_name_', number),
+        email='{}{}'.format('email_', number),
+        password='{}{}'.format('password_', number),
+    )
+
+
+def create_item(id_item, number):
+    return Item.create(
+        uuid=id_item,
+        name='{}{}'.format('name_', number),
+        price=number,
+        description='{}{}'.format('description_', number),
+        category='{}{}'.format('category_', number),
+    )
+
+
+class TestFavorites:
+    @classmethod
+    def setup_class(cls):
+        db = SqliteDatabase(':memory:')
+        tables = [Favorites, Item, User]
+        for table in tables:
+            table._meta.database = db
+            table.create_table()
+
+        cls.app = app.test_client()
+
+    def setup_method(self):
+        Item.delete().execute()
+        Favorites.delete().execute()
+        User.delete().execute()
+
+    def test_get__favorites(self):
+        id_user = uuid.uuid4()
+        id_item = uuid.uuid4()
+
+        user_db = create_user(
+            id_user,
+            1)
+
+        item_db = create_item(
+            id_item,
+            1)
+
+        Favorites.create(
+            uuid=uuid.uuid4(),
+            user=user_db,
+            item=item_db
+        )
+
+        resp = self.app.get('/favorites/')
+        assert resp.status_code == OK
+        data = json.loads(resp.data.decode())
+        assert len(data) == 1
+        assert user_db.favorite_items() == data
+
+    def test_get__favorites_empty(self):
+        resp = self.app.get('/favorites/')
+        data = json.loads(resp.data.decode())
+        assert not data
+        assert resp.status_code == NOT_FOUND
```

Comment:
????
se non ho preferiti perché devo avere un not_found? il not found indica che la risorsa non esiste, non che è vuota.